src/no_std.rs

```diff
@@ -0,0 +1,92 @@
+//! Contains logic related to no_std functionaligy
+
+use core::fmt;
+
+/// Avoids the use of [`std::io::Cursor`] in `no_std` context
+pub(crate) struct Cursor<T> {
+    buffer: T,
+    position: usize,
+}
+impl<T> Cursor<T> {
+    pub(crate) fn new(buffer: T) -> Self {
+        Self {
+            buffer,
+            position: 0,
+        }
+    }
+    pub(crate) fn position(&self) -> usize {
+        self.position
+    }
+}
+impl<T: AsMut<[u8]>> fmt::Write for Cursor<T> {
+    fn write_str(&mut self, s: &str) -> fmt::Result {
+        let raw_s = s.as_bytes();
+        let pos = self.position();
+
+        let buf = &mut self.buffer.as_mut()[pos..pos + raw_s.len()];
+
+        if raw_s.len() > buf.len() {
+            return Err(fmt::Error);
+        }
+
+        buf.copy_from_slice(raw_s);
+        self.position += raw_s.len();
+
+        Ok(())
+    }
+}
+
+#[cfg(any(test, feature = "std"))]
+fn f64_significant_digits_std(value: f64) -> usize {
+    (value.abs().log10() + 1f64).floor() as usize
+}
+
+#[cfg(any(test, not(feature = "std")))]
+fn f64_significant_digits_no_std(value: f64) -> usize {
```

Comment:
https://godbolt.org/z/fbdd5EP96

llvm.log10.f64 generates code without loops.

Anyway, we can always write a benchmark to compare implementations)

Reply:
Yeah, there won't be loops with `log10` but overall complexity will remain the same as the loop runs as many times as there are digits in the string decimal representation of the number

Reply:
> as many times as there are digits in the string decimal representation of the number

So, up to dozens of iterations at least?

Also, theoretically, can exist targets with hardware implementation, but I dunno such.

Reply:
Number of iterations depends on the number of digits in the integer part of the number

What I meant to say was that writing to string already has the same complexity so I don't expect this loop to degrade performance

Reply:
Yep, f64 → String is an expensive conversion even using `ryu`. However, I don't see reasons to use a custom unoptimized function without benches, even if this is only 5% difference.

Anywhere, the implementation with conversion sucks and definitely will be replaced with something more performant in the future.

Reply:
> Anyway, we can always write a benchmark to compare implementations)

I think it would only make sense to write benchmarks in the context of the whole crate together with parsing. Comparing just these two prob won't carry as much significance.

Reply:
Of course, `FixedPoint::try_from` and `FixedPoint::try_into` and compare run with and without std part.

Reply:
@mversic, merge as is, or want to bench it?

Reply:
Merge as is. Maybe I bench it at some point